### PR TITLE
Add opt-in Firefox listing metadata publishing

### PR DESCRIPTION
**Pull request by alphabt**
## Summary

- add a dependency-free, dry-run-first Node CLI that validates canonical store metadata and builds AMO's supported localized listing payload
- require `--apply` plus the existing Firefox issuer, secret, and add-on ID; generate a short-lived HS256 JWT and redact credentials from failures
- add an isolated manual workflow whose dry-run receives no secrets and whose apply mode requires typed confirmation
- document that Chrome/Edge and Firefox support/privacy fields remain dashboard-only

## Safety boundary

The PATCH contains only localized `name`, `summary`, `description`, and `homepage`, plus global `tags`. It does not touch packages, categories, screenshots, support URL, privacy policy, or automatic release publishing.

## Tests

- `npm test`
- `npm run format:check`
- `npm run zip`

### CODE REVIEW

File: scripts/publish-firefox-metadata.mjs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    throw new Error(
+      `--apply requires the following environment variables: ${missing.join(", ")}`,
+    );
+  }
+
+  return {
+    issuer: environment.FIREFOX_JWT_ISSUER,
+    secret: environment.FIREFOX_JWT_SECRET,
+    addonId: environment.FIREFOX_ADDON_ID.trim(),
+  };
```
[the remark this answers is not in the document]
Comment (alphabt) [reply scripts/publish-firefox-metadata.mjs:130]: Fixed in `24ae8cd`: `requireAmoCredentials()` now trims issuer and secret consistently with the add-on ID, and the regression test covers surrounding whitespace on all three values.


_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._
